ConstructBinaryTreefromInorderandPreorderTraversal.java

```diff
@@ -0,0 +1,47 @@
+package s30Coding;
+
+import java.util.*;
+
+//Time Complexity :- O(n) where n is number of elements in the list
+//Space Complexity :- O(h) where h is height of the tree
```

Comment:
Map is auxiliary space here, isn't it? And we are storing all the nodes with indices in map. So, O(N) space complexity